var/spack/repos/builtin/packages/py-cgen/package.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyCgen(PythonPackage):
+    """cgen offers a simple abstract syntax tree for C and related languages
+    (C++/CUDA/OpenCL) to allow structured code generation from Python.
+    """
+
+    homepage = "https://documen.tician.de/cgen/"
+    pypi     = "cgen/cgen-2020.1.tar.gz"
+
+    version('2020.1', sha256='4ec99d0c832d9f95f5e51dd18a629ad50df0b5464ce557ef42c6e0cd9478bfcf')
+
+    depends_on('python@:3.5,3.7:', type=('build', 'run'))
```

Comment:
Where do you see that Python 3.6 doesn't work?

Reply:
This was in the main brnach, removing.